src/java/org/apache/cassandra/db/rows/NativeCell.java

```diff
@@ -78,7 +78,7 @@ public NativeCell(NativeAllocator allocator,
         assert column.isComplex() == (path != null);
         if (path != null)
         {
-            assert path.size() == 1;
+            assert path.size() == 1 : String.format("Expected path size to be 1 but was not; %s", path);
```

Comment:
this is a nit from me; asserts should have messages to help (not needed for the patch, but is useful)